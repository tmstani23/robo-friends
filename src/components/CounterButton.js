import React, {PureComponent} from 'react';

//Pure components only update when their props change
class CounterButton extends PureComponent {
    
    state = {
        count: 0
    }
    // only necessary if not a pure component and you want to restrict rerenders.  Gives more directed control and allows updated based on specific props changes
    // shouldComponentUpdate(nextProps, nextState) {
    //     if (this.state.count !== nextState.count) {
    //         return true;
    //     }
    //     return false;
        
    // }

    updateCount = () => {
        //Here state is passed with a callback function to avoid side effects caused by async state update
        //Useful when modifying current state
        this.setState(state => {
            return {
                count: state.count + 1
            }
        })
    }

    render() {
        return (
            <button color={this.props.color} onClick={this.updateCount}>Count: {this.state.count}</button>    
        )
    }
}

export default CounterButton;