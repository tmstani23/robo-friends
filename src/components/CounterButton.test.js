import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';
import renderer from 'react-test-renderer';

it('expect card component to render', () => {
    const mockColor = 'red';
    
    const tree = renderer.create(
        <CounterButton color={mockColor} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
})

it('Test click event', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);
    const instance = wrapper.instance();
    expect(wrapper.state('count')).toBe(0);
    instance.updateCount();
    expect(wrapper.state('count')).toBe(1);
    expect(wrapper.props().color).toEqual('red')
});