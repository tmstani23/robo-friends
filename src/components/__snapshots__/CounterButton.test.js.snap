// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect card component to render 1`] = `
<button
  color="red"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
